### PR TITLE
RavenDB-14503 Handle stats for time-series

### PR DESCRIPTION
**Pull request by karmeli87**
- store the total count in a dedicated tree. So we can retrive it fast.
- lookup for the start/end date time

### CODE REVIEW

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs [hunk 1]
```diff
@@ -106,13 +110,111 @@ static TimeSeriesStorage()
...
+                {
+                    var segmentPtr = tvr.Read((int)TimeSeriesTable.Segment, out var size);
+                    var segment = new TimeSeriesValuesSegment(segmentPtr, size);
+
+                    var keyPtr = tvr.Read((int)TimeSeriesTable.TimeSeriesKey, out var keySize);
+                    
+                    dispose.Dispose();
+                    dispose = Slice.External(context.Allocator, keyPtr, keySize, out var keySlice);
+                    lastKey = keySlice;
+                    if (segment.NumberOfLiveEntries > 0)
```
Comment (ayende) [inline src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs]: What if I have 100000 segments with no live values?

Comment (karmeli87) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs]: It depends..
If they are deleted as a segment, we store only a single dead entry so each page will probably have several hundreds of those.. Which you still have to iterate.
If those segments are full with dead values, it is indeed a problem.

But in both cases after the cleaner did his job this shouldn't matter. 

I see 2 options:
1. We can do binary search (not sure this is a good idea.. the cost might be even higher)
2. Somehow track the first value

I don't see any nice/better way here.
I will test it on the 300M points data and see if it an issue.
If it isn't we leave this optimization for later. I think.

Comment (ayende) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs]: Alternatively, ignore dead values and report start /end as they are.
Then let the cleaner do its job?

Comment (karmeli87) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs]: Also an option

Comment (ayende) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs]: Let's go with this

Comment (karmeli87) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs]: done

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs [hunk 2]
```diff
@@ -106,13 +110,92 @@ static TimeSeriesStorage()
...
         {
             _documentDatabase = documentDatabase;
             _documentsStorage = documentDatabase.DocumentsStorage;
 
             tx.CreateTree(TimeSeriesKeysSlice);
             tx.CreateTree(DeletedRangesKey);
+            tx.CreateTree(TimeSeriesStats);
+        }
+
+        private void UpdateTotalCountStats(DocumentsOperationContext context, string docId, string name, long count)
```
Comment (ayende) [inline src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs:124]: I think that I'm missing calls to this from where we _delete_ values, no?

Comment (karmeli87) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs:124]: Upon every segment update, I reduce the old count and add the new count. This valid also for deletes. Even when we remove entire segment, we replace it with an empty one.. so the same logic apply as well.

**Review by ayende — APPROVED** [review]

**Comment by ppekrol** [comment]
We still need tests that cover Start/End dates.

**Pull request merged**